### PR TITLE
Feat: security improvements

### PR DESCRIPTION
**Pull request by andrewlew1s**
This is to add the authorisation headers as described in issue #30

### CODE REVIEW

File: orchestration-ui/src/store/modules/themes.js [hunk 1]
```diff
@@ -19,8 +19,11 @@ export default {
 	actions: {
 		async fetchTheme({ commit, rootGetters }, id) {
 			const { _id: networkId } = rootGetters['networks/network'];
+			const { token } = rootGetters.account.token;
 			const { data: theme } = await axios.get(
 				`${config.API_BASE}/themes/${id}`,
+				// this is where I added the header
+				{ headers: { Authorization: `Bearer ${token}` } },
```
Comment (michaelfitzhavey) [inline orchestration-ui/src/store/modules/themes.js]: This is great. We'll either need to add this to all the requests or do something like what this says: https://stackoverflow.com/questions/43051291/attach-authorization-header-for-all-axios-requests

If we do that we should probably create one shared instance of axios in store.js and share it similar to how we share the firebase/db object

**Comment by andrewlew1s** [comment]
updated themes, user and mecontrollers to how we did light controller. Didn't do networks as seems to do different stuff

File: orchestration-api/modules/me/MeController.js [hunk 2]
```diff
@@ -5,8 +5,7 @@ const router = Router();
 
 router.get('/', async (req, res, next) => {
 	try {
-		const { userId } = req.query;
-		const user = await getDocument(`users/${userId}`);
+		const user = req.user;
```
Comment (michaelfitzhavey) [inline orchestration-api/modules/me/MeController.js]: try replacing this with `{ user } = req;` (destructuring!)

File: orchestration-api/modules/users/UsersController.js [hunk 3]
```diff
@@ -51,8 +50,8 @@ router.post('/', async (req, res, next) => {
 
 router.patch('/:id', async (req, res, next) => {
 	try {
-		const { user, userId } = req.body;
-		const { networkId } = await getDocument(`users/${userId}`);
+		const { user } = req.body;
```
Comment (michaelfitzhavey) [inline orchestration-api/modules/users/UsersController.js]: in these cases let's make the updated user the body of the request (i.e. `user = req.body`). We'll need to change the format the request is sent in to be simply the user object: https://github.com/holistic-web/home-orchestrator/blob/feat/security-improvements/orchestration-ui/src/store/modules/users.js#L53

replace: `user: { userId, role }` with `userId, role`

File: orchestration-ui/src/router/index.js [hunk 4]
```diff
@@ -17,6 +18,7 @@ router.beforeEach((to, from, next) => {
 		return next();
 	}
 	if (!isAuthenticated) return next({ name: 'account.unauthorized' });
+	axios.defaults.headers.common.authorization = `Bearer ${store.getters['account/token']}`;
```
Comment (michaelfitzhavey) [inline orchestration-ui/src/router/index.js]: We can replace this with:
```
import firebase from 'firebase';
....
// the current file up to line 20
...
const token = await firebase.user.getIdToken();
axios.defaults.headers.common.authorization = `Bearer ${token}`;
```

This should ensure the token is refreshed whenever a user changes page (although the token may still expire if a user leaves a page open for an extended period of time), we'll fix that when we come to it

File: orchestration-api/modules/me/MeController.js (hunk 2 above)
Comment (andrewlew1s) [reply orchestration-api/modules/me/MeController.js]: ooh! fancy

File: orchestration-api/modules/users/UsersController.js (hunk 3 above)
Comment (andrewlew1s) [reply orchestration-api/modules/users/UsersController.js]: Ok, so does this currently mean if I'm not wrong:

user = req.body.user?

File: orchestration-ui/src/router/index.js (hunk 4 above)
Comment (andrewlew1s) [reply orchestration-ui/src/router/index.js]: done!

File: orchestration-api/modules/users/UsersController.js (hunk 3 above)
Comment (michaelfitzhavey) [reply orchestration-api/modules/users/UsersController.js]: if by 'this' you mean `{ user } = req.body`. Yes.

**Comment by michaelfitzhavey** [comment]
This looks awesome, have you checked if it works? 😄

**Comment by michaelfitzhavey** [comment]
Been + 3 weeks since an update so I'll carry this one home

**Comment by michaelfitzhavey** [comment]
This has now been deployed to live and seems to be working 🎉

**Pull request merged**